### PR TITLE
chore: update Scala parser and add in new query

### PR DESCRIPTION
**Pull request by ckipp01**
Note this can supersede https://github.com/nvim-treesitter/nvim-treesitter/pull/4174 since I also wanted to sync the queries.

### CODE REVIEW

File: queries/scala/highlights.scm [hunk 1]
```diff
@@ -96,6 +93,9 @@
 (generic_function
   function: (identifier) @function.call)
 
+(interpolated_string_expression
+  interpolator: (identifier) @function.call)
```
Comment (theHamsta) [inline queries/scala/highlights.scm:97]: Sting Interpolation is not a function call in other languages

Comment (ckipp01) [reply queries/scala/highlights.scm:97]: Yea, so I went back and forth about this. The main reason I chose this is that under the hood in Scala this is actually represented as a `def`. For example you can see this when hovering on the actual interpolator in Scala:

Screenshot 2023-01-18 at 09 25 49

It's a macro call, but I thought `function.call` was the most fitting here. Would you suggest something else?

Comment (theHamsta) [reply queries/scala/highlights.scm:97]: a it's the interpolator not the dollar style variable interpolation!

**Review by theHamsta — APPROVED** [review]

**Comment by clason** [comment]
don't merge master 😭

**Comment by theHamsta** [comment]
Wanted to squash the commit of the webui into the real one

**Pull request merged**